android/src/main/java/com/reactnativecommunity/netinfo/NetInfoUtils.java

```diff
@@ -25,4 +29,22 @@ public static boolean isAccessWifiStatePermissionGranted(Context context) {
         return ContextCompat.checkSelfPermission(context,
                 Manifest.permission.ACCESS_WIFI_STATE) == PackageManager.PERMISSION_GRANTED;
     }
+
+
+    /**
+     * Starting with Android 14, apps and services that target Android 14 and use context-registered
+     * receivers are required to specify a flag to indicate whether or not the receiver should be
+     * exported to all other apps on the device: either RECEIVER_EXPORTED or RECEIVER_NOT_EXPORTED
+     * <a href="https://developer.android.com/about/versions/14/behavior-changes-14#runtime-receivers-exported"/>
+     */
+    @SuppressLint("UnspecifiedRegisterReceiverFlag")
+    public static void compatRegisterReceiver(
+            Context context, BroadcastReceiver receiver, IntentFilter filter, boolean exported) {
+        if (Build.VERSION.SDK_INT >= 34 && context.getApplicationInfo().targetSdkVersion >= 34) {
+            context.registerReceiver(
+                    receiver, filter, exported ? Context.RECEIVER_EXPORTED : Context.RECEIVER_NOT_EXPORTED);
```

Comment:
This is a really minor point - because in all ways this looks like a nice way to handle it - just contemplating if it is possible to implement this without causing a breaking change (specifically: requiring the compile SDK33)

The constants here are defined and can be used directly (values 2 and 4) https://developer.android.com/reference/android/content/Context#RECEIVER_EXPORTED

I don't think there is a nice way around the API existing or not though, other than perhaps reflection - getting the method and invoking it.

So - and this is just thinking out loud - it seems like it *may* be possible to do this as a non-breaking change with hard coded constants and reflection API usage, but alternatively we could just say this is a breaking change and you need to be using compileSdk 33 minimum now.

I'm personally inclined to keep the code clean + easier to maintain and just issue the semver major with a nice developer note. 

What do you think?

Reply:
hi @mikehardy ty for taking a detailed look!

I definitely prefer not using reflection or magic numbers. +1 to merging as-is with compileSdk 33.